synapse/storage/databases/main/relations.py

```diff
@@ -671,12 +664,21 @@ async def get_bundled_aggregations(
         if not self._msc1849_enabled:
             return {}
 
-        # TODO Parallelize.
-        results = {}
+        # event ID -> bundled aggregation in non-serialized form.
+        results: Dict[str, Dict[str, Any]] = {}
+
+        event_ids = [event.event_id for event in events]
+
+        # Fetch any edits.
+        edits = await self._get_applicable_edits(event_ids)
+        for event_id, edit in edits.items():
+            results.setdefault(event_id, {})[RelationTypes.REPLACE] = edit
```

Comment:
I think the `setdefault` dance is used here to handle the case that an `event_id` occurs twice in `edits.items()`. Is that actually possible?

Reply:
No, it isn't necessary here, but is below when we call update. This was just me being dumb!

Reply:
Note that `dits` is a map so an `event_id` cannot occur twice.